Assets/UI/toppanel_CQUI.lua

```diff
@@ -9,9 +9,13 @@ BASE_CQUI_RefreshResources = RefreshResources;
 -- ===========================================================================
 local CQUI_showLuxury = true;
 
+function CQUI_OnSettingsInitialized()
+	CQUI_showLuxury = GameConfiguration.GetValue("CQUI_ShowLuxuries"); -- Infixo, issue #44
+end
+
 function CQUI_OnSettingsUpdate()
-  CQUI_showLuxury = GameConfiguration.GetValue("CQUI_ShowLuxuries");
-  RefreshResources();
+	CQUI_showLuxury = GameConfiguration.GetValue("CQUI_ShowLuxuries");
+	RefreshResources();
 end
```

Comment:
Any reason to add the new function? We register the same function for both `LuaEvents.CQUI_SettingsUpdate` and `LuaEvents.CQUI_SettingsInitialized` in most of the other places where we have both the events registered.

Reply:
As a general rule - each event should have its own hook function. Exceptions are always possible as usual (see the other issue - there was no need to add an extra hop). Here you can't make an exception because OnUpdate has an extra call (Refresh).